clang/lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -138,10 +135,7 @@ void visualstudio::Linker::ConstructJob(Compilation &C, const JobAction &JA,
       Args.hasArg(options::OPT_fsycl_host_compiler_EQ)) {
     CmdArgs.push_back(Args.MakeArgString(std::string("-libpath:") +
                                          TC.getDriver().Dir + "/../lib"));
-    if (Args.hasArg(options::OPT_g_Flag))
-      CmdArgs.push_back("-defaultlib:sycl" SYCL_MAJOR_VERSION "d.lib");
-    else
-      CmdArgs.push_back("-defaultlib:sycl" SYCL_MAJOR_VERSION ".lib");
+    CmdArgs.push_back("-defaultlib:sycl" SYCL_MAJOR_VERSION ".lib");
```

Comment:
Yes but in the case of the CMake module it uses `-nostdlib` before passing `-Xclang --dependent-lib=msvcrtd` so  `-defaultlib:msvcrt` doesn't get added in the toolchain.

The SYCL library currently doesn't respect `-nostdlib` (see: https://github.com/intel/llvm/pull/6699 ), so if we implicitly add `--dependent-lib=sycld` we also need to avoid adding `-defaultlib:sycld`.